project/data_analysis/python/get_xfreq_spectra.py

```diff
@@ -16,105 +16,179 @@
+                                                          spectra_order=modes_for_xar_cov,
+                                                          remove_doublon=True,
+                                                          check_pos_def=True)
 
-block_order["xfreq"] = {}
-block_order["xfreq"]["TT"] = fpairs
-block_order["xfreq"]["TE"] = fpairs_cross
-block_order["xfreq"]["EE"] = fpairs
+inv_cov["xar"] = np.linalg.inv(cov["xar"])
 
-block_order["final"] = {}
-block_order["final"]["TT"] = fpairs
-block_order["final"]["TE"] = ["all"]
-block_order["final"]["EE"] = ["all"]
+vec["xar"] = covariance.read_x_ar_spectra_vec(spec_dir,
+                                              spec_name_list,
+                                              "cross",
+                                              spectra_order = modes_for_xar_cov,
+                                              type="Dl")
+
+P_mat = covariance.get_x_ar_to_x_freq_P_mat(x_ar_cov_list,
+                                            x_freq_cov_list,
+                                            binning_file,
+                                            lmax)
+                                            
+cov["xfreq"] = covariance.get_max_likelihood_cov(P_mat,
+                                                 inv_cov["xar"],
+                                                 force_sim = True,
+                                                 check_pos_def = True)
+
+inv_cov["xfreq"] = np.linalg.inv(cov["xfreq"])
+
+vec["xfreq"]= covariance.max_likelihood_spectra(cov["xfreq"],
+                                                inv_cov["xar"],
+                                                P_mat,
+                                                vec["xar"])
+
+P_final = covariance.get_x_freq_to_final_P_mat(x_freq_cov_list,
+                                               final_cov_list,
+                                               binning_file,
+                                               lmax)
+                                               
+cov["final"]  = covariance.get_max_likelihood_cov(P_final,
+                                                  inv_cov["xfreq"],
+                                                  force_sim = True,
+                                                  check_pos_def = True)
+                                                  
+inv_cov["final"] = np.linalg.inv(cov["final"])
+
+vec["final"] = covariance.max_likelihood_spectra(cov["final"],
+                                                inv_cov["xfreq"],
+                                                P_final,
+                                                vec["xfreq"])
+
+
+covariance.plot_P_matrix(P_mat, x_freq_cov_list, x_ar_cov_list, file_name=f"{combine_dir}/P_mat_x_ar_to_x_freq")
+covariance.plot_P_matrix(P_final, final_cov_list, x_freq_cov_list, file_name=f"{combine_dir}/P_mat_x_freq_to_final")
 
-combin_level = ["xfreq", "final"]
 for comb in combin_level:
```

Comment:
I don't see the need for `combin_level` just do `for comb, matrix in cov.items()`